src/components/config/probe.ts

```diff
@@ -0,0 +1,43 @@
+import type { Probe } from '../../interfaces/probe'
+
+const probes: Map<string, Probe> = new Map()
+
+export function getProbes() {
+  const legacyProbes: Probe[] = []
```

Comment:
It is called legacy because it is. In the future it will use [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), as it is more performant and easier to use for probe data. Will remove the legacy array in the the different PR.